portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -195,26 +220,83 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	});
 	return def.promise();
     };
+    var getCOSMICCounts = function(webservice_data) {
+	/* In: - webservice_data, a list of data obtained from the webservice API
+	 * Out: Promise which resolves with map from keyword to COSMIC count records
+	 */
+	var def = new $.Deferred();
+	var keywords = webservice_data.filter(function(datum) { return datum.genetic_alteration_type === "MUTATION_EXTENDED" && typeof datum.keyword !== 'undefined' && datum.keyword !== null; })
+					.map(function(mutation_datum_with_keyword) { return mutation_datum_with_keyword.keyword; });
+	var counts = {};
+	$.ajax({
+	    type: 'POST',
+	    url: 'api/cosmic_count',
+	    data: 'keywords='+keywords.join(",")
+	}).then(function(cosmic_count_records) {
+	    for (var i=0; i<cosmic_count_records.length; i++) {
+		var keyword = cosmic_count_records[i].keyword;
+		counts[keyword] = counts[keyword] || [];
+		counts[keyword].push(cosmic_count_records[i]);
+	    }
+	    def.resolve(counts);
+	});
+	return def.promise();
+    };
     var getOncoKBAnnotations = function(webservice_data) {
-	var is_oncogenic = {}; // #{gene}&#{alteration.toUpperCase()}&{tumor_type.toUpperCase()} is key, boolean is value
+	/* In: - webservice_data, a list of data obtained from the webservice API
+	 * Out: Promise which resolves with map from gene.toUpperCase() to amino acid change.toUpperCase() to one of ['Unknown', 'Likely Neutral', 'Likely Oncogenic', 'Oncogenic']
+	 */
+	var def = new $.Deferred();
+	var oncogenic = {}; // See Out above
+	
+	// Collect genes and alterations to query
+	for (var i=0; i<webservice_data.length; i++) {
+	    var datum = webservice_data[i];
+	    if (datum.genetic_alteration_type === "MUTATION_EXTENDED") {
+		var gene = datum.hugo_gene_symbol.toUpperCase();
+		var alteration = datum.amino_acid_change.toUpperCase();
+		oncogenic[gene] = oncogenic[gene] || {};
+		oncogenic[gene][alteration] = false;
+	    }
+	}
+	var queries = [];
+	var query_genes = Object.keys(oncogenic);
+	for (var i=0; i<query_genes.length; i++) {
+	    var query_alterations = Object.keys(oncogenic[query_genes[i]]);
+	    for (var j=0; j<query_alterations.length; j++) {
+		queries.push({'hugoSymbol': query_genes[i], 'alteration': query_alterations[j]});
+	    }
+	}
+	// Execute query
 	var query = {
 	    "geneStatus": "Complete",
 	    "source": "cbioportal",
 	    "evidenceTypes": "GENE_SUMMARY,GENE_BACKGROUND,ONCOGENIC,MUTATION_EFFECT,VUS,STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY,STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_RESISTANCE,INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY",
```

Comment:
Could you change evidenceTypes to 'ONCOGENIC' since you only need oncogenicity info？